heartbeat/NVMePort

```diff
@@ -0,0 +1,228 @@
+<longdesc lang="en">
+The NVMe transport type. Might be one of 'tcp', 'rdma', 'fc' or 'loop'.
+</longdesc>
+<shortdesc lang="en">tcp or rdma</shortdesc>
+<content type="string" default="${OCF_RESKEY_type_default}"/>
+</parameter>
+
+<parameter name="addr_fam" required="0" unique="0">
+<longdesc lang="en">
+The address family of the address. Mibht be 'ipv4', 'ipv6' or 'fc'.
+</longdesc>
+<shortdesc lang="en">ipv4 or ipv6</shortdesc>
+<content type="string" default="${OCF_RESKEY_addr_fam_default}"/>
+</parameter>
+
+<parameter name="svcid" required="0" unique="0">
+<longdesc lang="en">
+The transport service identifier, is the TCP/IP port number of the
+equivalent in the RDMA world.
+</longdesc>
+<shortdesc lang="en">IP/RDMA port number</shortdesc>
+<content type="integer" default="${OCF_RESKEY_svcid_default}"/>
+</parameter>
+
+<parameter name="addr" required="1" unique="0">
+<longdesc lang="en">
+The transport address, is the TCP/IP address this targets binds to
+or the equivalent in the RDMA world.
+</longdesc>
+<shortdesc lang="en">IP/RDMA/FC address</shortdesc>
+<content type="string"/>
+</parameter>
+
+<parameter name="nqns" required="1" unique="0">
+<longdesc lang="en">
+A space-separated list of NQNs that should be exported through
+this NVMe-oF-Target port. This list needs to have at least one entry.
+</longdesc>
+<shortdesc lang="en">list of NQNs</shortdesc>
+<content type="string"/>
+</parameter>
+
+</parameters>
+
+<actions>
+<action name="start"		timeout="10s" />
+<action name="stop"		 timeout="10s" />
+<action name="status"	   timeout="10s" interval="10s" depth="0" />
+<action name="monitor"	  timeout="10s" interval="10s" depth="0" />
+<action name="meta-data"	timeout="5s" />
+<action name="validate-all"   timeout="10s" />
+</actions>
+</resource-agent>
+END
+}
+
+#######################################################################
+
+NVMePort_usage() {
+	cat <<END
+usage: $0 {start|stop|status|monitor|validate-all|meta-data}
+
+Expects to have a fully populated OCF RA-compliant environment set.
+END
+}
+
+NVMePort_start() {
+	NVMePort_monitor
+	if [ $? =  $OCF_SUCCESS ]; then
+		return $OCF_SUCCESS
+	fi
+
+	mkdir ${portdir}
+	echo ${OCF_RESKEY_addr} > ${portdir}/addr_traddr
+	echo ${OCF_RESKEY_type} > ${portdir}/addr_trtype
+	echo ${OCF_RESKEY_svcid} > ${portdir}/addr_trsvcid
+	echo ${OCF_RESKEY_addr_fam} > ${portdir}/addr_adrfam
+
+	for subsystem in ${OCF_RESKEY_nqns}; do
+		ln -s /sys/kernel/config/nvmet/subsystems/${subsystem} \
+		   ${portdir}/subsystems/${subsystem}
+	done
+
+	NVMePort_monitor
+}
+
+NVMePort_stop() {
+	NVMePort_monitor
+	if [ $? -eq $OCF_NOT_RUNNING ]; then
+		return $OCF_SUCCESS
+	fi
+
+	for subsystem in ${OCF_RESKEY_nqns}; do
+		rm ${portdir}/subsystems/${subsystem}
+	done
+
+	rmdir ${portdir}
+
+	return $OCF_SUCCESS
+}
+
+NVMePort_monitor() {
+	[ -d ${portdir} ] || return $OCF_NOT_RUNNING
+	return $OCF_SUCCESS
+}
+
+NVMePort_validate() {
+	if [[ ! ${OCF_RESKEY_type} =~ ^(tcp|rdma|fc)$ ]]; then
```

Comment:
If you replace these with case-statements you can update the `#!` to sh, which is preferred, and avoid renaming them to .in files and adding to configure.